bin/varnishd/http2/cache_http2_proto.c

```diff
@@ -218,10 +219,16 @@ h2_del_req(struct worker *wrk, struct h2_req **pr2)
 		AZ(stvbuf);
 	}
 
-	Req_Cleanup(sp, wrk, r2->req);
+	req = r2->req;
+	CHECK_OBJ_NOTNULL(req, REQ_MAGIC);
+	r2->magic = 0;
+	req->transport_priv = NULL;
```

Comment:
I believe these two lines need to be moved after `Req_Cleanup()`. This due to `Req_Cleanup()` calling `VDP_Fini()`, which I guess may for some VDP call on code expecting to see the complete structure.

Reply:
Good catch, I forgot about the increased scope of `Req_Cleanup()` in this branch.